src/UI/Titlebar.re

```diff
@@ -7,57 +7,252 @@ open Revery.UI;
+      height(30),
+      backgroundColor(
+        isFocused
+          ? Colors.activeBackground.from(theme)
+          : Colors.inactiveBackground.from(theme),
+      ),
+      flexDirection(`Row),
+      justifyContent(`SpaceBetween),
+      alignItems(`Center),
+    ];
+
+    let iconAndTitle = [
+      flexDirection(`Row),
+      alignItems(`Center),
+      marginHorizontal(8),
+    ];
+
+    let title = (~isFocused, ~theme, ~font: UiFont.t) => [
+      flexGrow(0),
+      fontSize(12.),
+      fontFamily(font.fontFile),
+      marginLeft(4),
+      backgroundColor(
+        isFocused
+          ? Colors.activeBackground.from(theme)
+          : Colors.inactiveBackground.from(theme),
+      ),
+      color(
+        isFocused
+          ? Colors.activeForeground.from(theme)
+          : Colors.inactiveForeground.from(theme),
+      ),
+      textWrap(TextWrapping.NoWrap),
+    ];
+
+    let buttons = [flexDirection(`Row), alignItems(`Center)];
+
+    module Button = {
+      let container = [
+        height(30),
+        width(46),
+        justifyContent(`Center),
+        alignItems(`Center),
+      ];
+
+      let hoverClose = (~theme) => [
+        height(30),
+        width(46),
+        backgroundColor(Colors.hoverCloseBackground.from(theme)),
+        justifyContent(`Center),
+        alignItems(`Center),
+      ];
+
+      let hover = (~theme) => [
+        height(30),
+        width(46),
+        backgroundColor(Colors.hoverBackground.from(theme)),
+        justifyContent(`Center),
+        alignItems(`Center),
+      ];
+    };
+  };
 };
 
-let make = (~isFocused, ~isFullscreen, ~title, ~theme, ~font: UiFont.t, ()) =>
-  if (isFullscreen) {
-    React.empty;
-  } else {
-    switch (Revery.Environment.os) {
-    | Mac =>
+module Mac = {
+  let make =
+      (~isFocused, ~windowDisplayMode, ~title, ~theme, ~font: UiFont.t, ()) =>
+    if (windowDisplayMode == Model.State.Fullscreen) {
+      React.empty;
+    } else {
       <Clickable
         onDoubleClick={_ =>
           GlobalContext.current().dispatch(Model.Actions.TitleDoubleClicked)
         }
-        style={Styles.container(~isFocused, ~theme)}>
-        <Text style={Styles.text(~isFocused, ~theme, ~font)} text=title />
-      </Clickable>
-    | _ => React.empty
+        style={Styles.Mac.container(~isFocused, ~theme)}>
+        <Text style={Styles.Mac.text(~isFocused, ~theme, ~font)} text=title />
+      </Clickable>;
+    };
+};
+
+module Windows = {
+  module Buttons = {
+    module Close = {
+      let%component make = (~theme, ()) => {
+        let%hook (isHovered, setHovered) = Hooks.state(false);
+
+        let onMouseUp = _ =>
+          GlobalContext.current().dispatch(Model.Actions.WindowCloseClicked);
+
+        <View
+          style={
+            isHovered
+              ? Styles.Windows.Button.hoverClose(theme)
+              : Styles.Windows.Button.container
+          }
+          onMouseUp
+          onMouseEnter={_ => setHovered(_ => true)}
```

Comment:
I like that you added the hover styles, awesome!

Reply:
Is this default behavior on your system? On stock apps on my VM the cursor is just the standard pointer. My VM has a bunch of weird quirks though so it's entirely possible it's just mine.

Reply:
You're right, I double-checked on my non-VM-Windows machine, and actually this would differ from the default behavior - thanks for checking! So we don't need to add cursor styles.